workers/unity/Packages/com.improbable.gdk.querybasedinteresthelper/InterestTemplate.cs

```diff
@@ -0,0 +1,335 @@
+using System.Collections.Generic;
+using Improbable.Gdk.Core;
+
+namespace Improbable.Gdk.QueryBasedInterest
+{
+    /// <summary>
+    ///     Utility class to help construct Interest component snapshots.
+    /// </summary>
+    public class InterestTemplate
+    {
+        private readonly Dictionary<uint, ComponentInterest> interest;
+
+        private InterestTemplate(Dictionary<uint, ComponentInterest> interest = null)
+        {
+            this.interest = interest ?? new Dictionary<uint, ComponentInterest>();
+        }
+
+        /// <summary>
+        ///     Creates a new InterestTemplate object.
+        /// </summary>
+        /// <returns>
+        ///     A new InterestTemplate object.
+        /// </returns>
+        public static InterestTemplate Create()
+        {
+            return new InterestTemplate();
+        }
+
+        /// <summary>
+        ///     Creates a new InterestTemplate object given an existing InterestTemplate.
+        /// </summary>
+        /// <param name="interest">
+        ///     An existing InterestTemplate.
+        /// </param>
+        /// <returns>
+        ///     An InterestTemplate object.
+        /// </returns>
+        public static InterestTemplate Create(InterestTemplate interestTemplate)
```

Comment:
As discussed we should rename these to `CopyFrom` or similar to make the intent explicit.